boudicca.events/eventcollectors/src/main/kotlin/events/boudicca/eventcollector/collectors/FraeuleinFlorentineCollector.kt

```diff
@@ -0,0 +1,97 @@
+package events.boudicca.eventcollector.collectors
+
+import base.boudicca.SemanticKeys
+import base.boudicca.api.eventcollector.TwoStepEventCollector
+import base.boudicca.api.eventcollector.util.FetcherFactory
+import base.boudicca.format.UrlUtils
+import base.boudicca.model.structured.StructuredEvent
+import base.boudicca.model.structured.dsl.structuredEvent
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+
+
+class FraeuleinFlorentineCollector : TwoStepEventCollector<Element>("fraeuleinflorentine") {
+
+    private val fetcher = FetcherFactory.newFetcher()
+    private val baseUrl = "https://frl-florentine.at/eventkalender/"
+
+    override fun getAllUnparsedEvents(): List<Element> {
+        return Jsoup.parse(fetcher.fetchUrl(baseUrl))
+            .select("ul.simcal-events li")
+            .distinctBy { it.text() } // multi-day events have the same text in multiple entries
+    }
+
+    override fun parseStructuredEvent(event: Element): StructuredEvent {
+        val nameAndTime = event.select(".simcal-event-title").text().split("|")
+        val name = nameAndTime.first().trim()
+        val description = event.select(".simcal-event-description").text()
+        val startDate = parseStartDate(event, nameAndTime)
+        val endDate = parseEndDate(event, startDate)
+
+        return structuredEvent(name, startDate) {
+            withProperty(SemanticKeys.URL_PROPERTY, UrlUtils.parse(baseUrl))
+            withProperty(SemanticKeys.SOURCES_PROPERTY, listOf(baseUrl))
+            withProperty(SemanticKeys.DESCRIPTION_TEXT_PROPERTY, description)
+            withProperty(SemanticKeys.ENDDATE_PROPERTY, endDate)
+            withProperty(SemanticKeys.LOCATION_NAME_PROPERTY, "Salonschiff Fräulein Florentine ")
+            withProperty(SemanticKeys.LOCATION_URL_PROPERTY, UrlUtils.parse("https://frl-florentine.at"))
+            withProperty(SemanticKeys.LOCATION_CITY_PROPERTY, "Linz")
+        }
+    }
+
+    private fun parseStartDate(event: Element, nameAndTime: List<String?>): OffsetDateTime {
+        var startTimeToParse = "00:00 AM"
```

Comment:
why did you not use the new dateparser?

Reply:
oh but i see that they use AM/PM as well <.< this does not work yet, so just leave it as it for now